index.js

```diff
@@ -209,6 +213,60 @@ app.get("/popup", function (req, res) {
   res.status(200).send(popupMsg);
 });
 
+const modifyCoordinates = originalResponse =>
+  originalResponse.reduce((acc, curr) => {
+    const {
+      coordinates: {
+        coordinates: [longitude, latitude]
+      }
+    } = curr;
+
+    return [
+      ...acc,
+      {
+        ...curr,
+        coordinates: {
+          latitude,
+          longitude
+        }
+      }
+    ];
+  }, []);
+
+app.use(
+  "/food",
+  createProxyMiddleware({
+    target: config.get("foodUrl"),
+    changeOrigin: true,
+    pathRewrite: { "^/food": "" },
+    selfHandleResponse: true,
+    onProxyRes: responseInterceptor(async responseBuffer => {
```

Comment:
Please check for response Content-Type and just return the original buffer if it's not `application/json`.
`if (proxyRes.headers['content-type']?.includes('application/json')) { ... }`

This is like https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/response-interceptor.md#manipulate-json-responses-applicationjson, but content type header could include more than just `application/json`. In this case, it's `application/json;charset=utf-8`

Reply:
ok